src/client/lib/hooks/useCmpConsent.ts

```diff
@@ -3,55 +3,23 @@ import { ConsentState } from '@guardian/consent-management-platform/dist/types';
 import { onConsentChange } from '@guardian/consent-management-platform';
 
 /**
- * This hook checks the CMP (Consent Management Platform) browser cookies. It returns true only if everything
- * has been consented to. This code is lifted from the Sign In Gate Display Rules in DotCom-Rendering repo:
- * https://github.com/guardian/dotcom-rendering/blob/main/dotcom-rendering/src/web/components/SignInGate/displayRule.ts#L71-L105
+ * ConsentState.canTarget checks if user has consented to all consents
+ * https://github.com/guardian/consent-management-platform#consentstatecantarget
  */
-const hasRequiredConsents = (): Promise<boolean> => {
-  const hasConsentedToAll = (state: ConsentState) => {
-    const consentFlags = state.tcfv2?.consents
-      ? Object.values(state.tcfv2.consents)
-      : [];
-    const vendorConsentFlags = state.tcfv2?.vendorConsents
-      ? Object.values(state.tcfv2.vendorConsents)
-      : [];
-    const isEmpty =
-      consentFlags.length === 0 || vendorConsentFlags.length === 0;
-
-    return !isEmpty && [...consentFlags, ...vendorConsentFlags].every(Boolean);
-  };
-
+const hasRequiredCmpConsents = (): Promise<boolean> => {
   return new Promise((resolve) => {
-    onConsentChange((state) => {
-      if (state.tcfv2) {
-        return resolve(hasConsentedToAll(state));
-      }
-
-      if (state.ccpa) {
-        return resolve(state.ccpa.doNotSell === false);
-      }
-
-      if (state.aus) {
-        return resolve(state.aus.personalisedAdvertising);
-      }
-
-      // this shouldn't ever be hit, but this is here as safety
-      return resolve(false);
+    onConsentChange((state: ConsentState) => {
+      return resolve(state.canTarget);
```

Comment:
This isn't going to cause any problems, but I thought I'd mention that you don't really need to return from within a Promise, I expect the previous code was just returning as a way to break out of the function.